cmd/zfs/zfs_main.c

```diff
@@ -5238,10 +5238,17 @@ parse_fs_perm(fs_perm_t *fsperm, nvlist_t *nvl)
 					break;
 				}
 
-				if (nice_name != NULL)
+				if (nice_name != NULL) {
 					(void) strlcpy(
 					    node->who_perm.who_ug_name,
 					    nice_name, 256);
+				} else {
+					/* User or group unknown */
+					(void) snprintf(
+					    node->who_perm.who_ug_name,
+					    sizeof (node->who_perm.who_ug_name),
+					    "(unknown: %d)", rid);
```

Comment:
I know some similar tools (e.g. `ls -l`) will print only the ID if it can't be translated to a name.  e.g. `"%d"` Instead of `"(unknown: %d)"`.  One advantage is that it is more parseable since the format is more consistent.  Not sure how much of a consideration that is here, but wanted to get your thoughts.

Reply:
I don't really have an opinion on this.
On the one hand, the suggested uniformity looks nicer.
On the other hand, at least FreeBSD allows to create user/group names consisting only of digit symbols, for example:
```
202020:*:202020:202020:Test Tester:/home/202020:/bin/sh

$ id
uid=202020(202020) gid=202020(202020) groups=202020(202020)
```
So, potentially I could create a user with name "1000" and ID 1001 and another user with name "1001" and ID 1000 and then watch the resulting confusion :smiley:
So, `"(unknown: %d)"` allows to be sure that the numeric user ID is printed and that the user name is printed in the other case.

Reply:
How about using a consistent format here, "%s(%d)", and just leave the string part empty if unknown?

Reply:
Is '?' a valid username on any of the potential platforms?